recipes-bsp/boot0/boot0.bb

```diff
@@ -0,0 +1,39 @@
+DESCRIPTION = "Mainline friendly Secondary Program Loader for Allwinner D1"
+LICENSE = "GPL-2.0-or-later"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/GPL-2.0-or-later;md5=fed54355545ffd980b814dab4a3b312c"
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+SRC_URI = " \
+	git://github.com/smaeul/sun20i_d1_spl;protocol=https;branch=mainline \
+	file://0001-config.mk-provide-path-to-linux-headers.patch \
+	file://0002-riscv-fix-build-with-binutils-2.38.patch \
+"
+
+PV = "1.0-dev"
+SRCREV = "0ad88bfdb723b1ac74cca96122918f885a4781ac"
+
+S = "${WORKDIR}/git"
+
+CFLAGS += " --sysroot=${RECIPE_SYSROOT}"
```

Comment:
--sysroot is already passed by TOOLCHAIN_OPTIONS var which is appended to CC var already, so perhaps you need to adjust makefile to allow it use CC from env

Reply:
I apply your suggestion with an updated patch:
```
recipes-bsp/boot0/files/0001-config.mk-provide-path-to-linux-headers.patch
```